src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -1644,4 +1736,77 @@ public void setParents(final String firstParent, final String secondParent)
     {
         this.parents = new Tuple<>(firstParent, secondParent);
     }
+
+    @Override
+    public void setIdleDays(final int days)
+    {
+
+    }
+
+    @Override
+    public void assignQuest(final IColonyQuest quest)
+    {
+        this.availableQuests.add(quest.getId());
+    }
+
+    @Override
+    public void openDialogue(final IColonyQuest quest, final int index)
+    {
+        final Component comp = Component.literal(quest.getId().toString());
```

Comment:
Yep, but that would be a non backwards compatible change to the system. So, for now we have to do this.

Reply:
Why does that matter?  Change the legacy ones first (separate PR if needed).  Nothing outside of mcol itself depends on it (since it can't) so you don't risk breaking anything.  The only possible risk is for serialization, but I wouldn't have thought these were currently serialized.

Reply:
They are serialized, thats what I meant